oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java

```diff
@@ -59,12 +59,18 @@ public final class OidcProviderConfigurationEndpointFilter extends OncePerReques
 
 	private final ProviderSettings providerSettings;
 	private final RequestMatcher requestMatcher;
+	private final boolean dynamicClientRegistrationEnabled;
 	private final OidcProviderConfigurationHttpMessageConverter providerConfigurationHttpMessageConverter =
 			new OidcProviderConfigurationHttpMessageConverter();
 
 	public OidcProviderConfigurationEndpointFilter(ProviderSettings providerSettings) {
+		this(providerSettings, false);
+	}
+
+	public OidcProviderConfigurationEndpointFilter(ProviderSettings providerSettings, boolean dynamicClientRegistrationEnabled) {
```

Comment:
Adding the flag `dynamicClientRegistrationEnabled` to the constructor will not provide us the flexibility we need. There will be other features that are enabled/disabled and adding more flags to constructor provides limited flexibility and API will constantly change.

Reply:
Hi @jgrandja,
Thanks for your feedback.
One thing we can do is keep this flag inside provider settings as this is a provider related settings. What do you think? Or do you have any better idea for this keeping this flag?

Reply:
Thanks for your feed, will try to find a way out

Reply:
Hi @jgrandja,
I have updated the PR. Please have a look and provider you feedback.